File: test/inspect_db.dart
import 'package:supabase_flutter/supabase_flutter.dart';
import 'package:flutter_dotenv/flutter_dotenv.dart';
import 'dart:io';

void main() async {
  // Load environment variables
  await dotenv.load();
  final supabaseUrl = dotenv.env['SUPABASE_URL'] ?? '';
  final supabaseAnonKey = dotenv.env['SUPABASE_ANON_KEY'] ?? '';

  print('Initializing Supabase client...');
  final client = SupabaseClient(supabaseUrl, supabaseAnonKey);

  print('Querying users...');
  try {
    final users = await client.from('users').select('id, email, role, full_name');
    print('Users count: ${users.length}');
    for (var u in users) {
      print('User: ${u['id']} - ${u['email']} - ${u['role']} - ${u['full_name']}');
    }
  } catch (e) {
    print('Error querying users: $e');
  }

  print('\nQuerying students...');
  try {
    final students = await client.from('students').select();
    print('Students: $students');
  } catch (e) {
    print('Error querying students: $e');
  }

  print('\nQuerying subscriptions...');
  try {
    final subs = await client.from('subscriptions').select();
    print('Subscriptions: $subs');
  } catch (e) {
    print('Error querying subscriptions: $e');
  }

  print('\nQuerying payments...');
  try {
    final payments = await client.from('payments').select();
    print('Payments: $payments');
  } catch (e) {
    print('Error querying payments: $e');
  }

  print('\nQuerying exams...');
  try {
    final exams = await client.from('exams').select('id, title, is_published, start_at, end_at');
    print('Exams count: ${exams.length}');
    for (var ex in exams) {
      print('Exam: ${ex['id']} - ${ex['title']} - published: ${ex['is_published']} - start: ${ex['start_at']} - end: ${ex['end_at']}');
    }
  } catch (e) {
    print('Error querying exams: $e');
  }

  exit(0);
}

File: test/inspect_db.py
import os
import urllib.request
import json

def get_supabase_data(url, key, table):
    headers = {
        'apikey': key,
        'Authorization': f'Bearer {key}'
    }
    req = urllib.request.Request(f"{url}/rest/v1/{table}?select=*", headers=headers)
    try:
        with urllib.request.urlopen(req) as response:
            return json.loads(response.read().decode())
    except Exception as e:
        print(f"Error reading {table}: {e}")
        return []

def main():
    # Load .env manually
    env = {}
    if os.path.exists('.env'):
        with open('.env') as f:
            for line in f:
                if '=' in line:
                    k, v = line.strip().split('=', 1)
                    env[k] = v

    url = env.get('SUPABASE_URL', '')
    key = env.get('SUPABASE_ANON_KEY', '')

    if not url or not key:
        print("Missing SUPABASE_URL or SUPABASE_ANON_KEY")
        return

    print("--- USERS ---")
    users = get_supabase_data(url, key, 'users')
    for u in users:
        print(f"ID: {u.get('id')} | Email: {u.get('email')} | Role: {u.get('role')} | Name: {u.get('full_name')}")

    print("\n--- STUDENTS ---")
    students = get_supabase_data(url, key, 'students')
    for s in students:
        print(f"ID: {s.get('id')} | Grade: {s.get('grade_level')} | Parent Phone: {s.get('parent_phone')}")

    print("\n--- SUBSCRIPTIONS ---")
    subs = get_supabase_data(url, key, 'subscriptions')
    for sub in subs:
        print(f"ID: {sub.get('id')} | Student: {sub.get('student_id')} | Teacher: {sub.get('teacher_id')} | Status: {sub.get('status')}")

    print("\n--- PAYMENTS ---")
    payments = get_supabase_data(url, key, 'payments')
    for p in payments:
        print(f"ID: {p.get('id')} | Payer: {p.get('payer_id')} | Course: {p.get('course_id')} | Amount: {p.get('amount')} | Gateway: {p.get('payment_gateway')} | Status: {p.get('status')}")

    print("\n--- EXAMS ---")
    exams = get_supabase_data(url, key, 'exams')
    for ex in exams:
        print(f"ID: {ex.get('id')} | Title: {ex.get('title')} | Teacher: {ex.get('teacher_id')} | Published: {ex.get('is_published')} | Start: {ex.get('start_at')} | End: {ex.get('end_at')}")

if __name__ == '__main__':
    main()
